Source/SwiftLintBuiltInRules/Rules/Style/FunctionNameWhitespaceRule.swift

```diff
@@ -0,0 +1,67 @@
+import SwiftSyntax
+
+@SwiftSyntaxRule(correctable: true, optIn: true)
+struct FunctionNameWhitespaceRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "function_name_whitespace",
+        name: "Function Name Whitespace",
+        description: "Function declarations must have exactly one space before the name and no space after it",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("func abc(lhs: Int, rhs: Int) -> Int {}"),
+        ],
+        triggeringExamples: [
+            Example("func  name(lhs: A, rhs: A) -> A {}"),       // 2 spaces before
+            Example("func ↓name (lhs: A, rhs: A) -> A {}"),      // 1 space after
+            Example("↓func  ↓name (lhs: A, rhs: A) -> A {}"),    // 1 space after, 2 spaces before
+        ],
+        corrections: [
+            Example("↓func  ↓name (lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+            Example("↓func  name(lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+            Example("↓func   name(lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+            Example("func ↓name (lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
```

Comment:
We don't need the violation indicators in correction examples:
```suggestion
            Example("func  name (lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
            Example("func  name(lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
            Example("func   name(lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
            Example("func name (lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
```